src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +40,317 @@ const ActorInit En_Door_InitVars = {
     (ActorFunc)EnDoor_Update,
     (ActorFunc)EnDoor_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Destroy.s")
+/**
+ * Controls which object and display lists to use in a given scene
+ */
+static EnDoorInfo sDoorInfo[] = {
+    { SCENE_HIDAN, 1, OBJECT_HIDAN_OBJECTS },
+    { SCENE_MIZUSIN, 2, OBJECT_MIZU_OBJECTS },
+    { SCENE_HAKADAN, 3, OBJECT_HAKA_DOOR },
+    { SCENE_HAKADANCH, 3, OBJECT_HAKA_DOOR },
+    // KEEP objects should remain last and in this order
+    { -1, 0, OBJECT_GAMEPLAY_KEEP },
+    { -1, 4, OBJECT_GAMEPLAY_FIELD_KEEP },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
+};
+
+AnimationHeader* D_809FCECC[] = { 0x0400E758, 0x0400E5B4, 0x04010038, 0x0400E6A0 };
+
+static u8 sDoorAnimOpenFrames[] = { 25, 25, 25, 25 };
+
+static u8 sDoorAnimCloseFrames[] = { 60, 70, 60, 70 };
+
+static Gfx* D_809FCEE4[5][2] = {
+    { 0x0400ECB8, 0x0400EE00 }, { 0x0600F998, 0x0600F938 }, { 0x06004958, 0x06004A10 },
+    { 0x060013B8, 0x06001420 }, { 0x050047A0, 0x05004978 },
+};
+
+extern Gfx D_0400ECB8[];
+extern Gfx D_0400EE00[];
+
+extern AnimationHeader D_0400E758;
+extern SkeletonHeader D_0400FF78;
+
+void EnDoor_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnDoor* this = THIS;
+    EnDoorInfo* objectInfo;
+    s32 i;
+    s32 objIndex;
+    f32 xOffset;
+    f32 zOffset;
+
+    objectInfo = &sDoorInfo[0];
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0400FF78, &D_0400E758, this->limbDrawTable,
+                   this->transitionDrawTable, 5);
+    for (i = 0; i < ARRAY_COUNT(sDoorInfo) - 2; i++, objectInfo++) {
+        if (globalCtx->sceneNum == objectInfo->sceneNum) {
+            break;
+        }
+    }
+    if (i >= ARRAY_COUNT(sDoorInfo) - 2 && Object_GetIndex(&globalCtx->objectCtx, OBJECT_GAMEPLAY_FIELD_KEEP) >= 0) {
+        objectInfo++;
+    }
+
+    this->displaylistIdx = objectInfo->displaylistIdx;
+    objIndex = Object_GetIndex(&globalCtx->objectCtx, objectInfo->objectId);
+    if (objIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->requiredObject = objIndex;
+    this->displaylistIdx = objectInfo->displaylistIdx;
+    if (this->actor.objBankIndex == this->requiredObject) {
```

Comment:
I suggest naming `requiredObject` with something that makes it clear it's a bank index and not an object id.
For instance:
```suggestion
    if (this->actor.objBankIndex == this->requiredObjBankIndex) {
```

Reply:
id also rename that variable above to `dListIndex` as we've been trying to make that abbreviation more consistent